### PR TITLE
Jupyter job status magic

### LINKED ISSUE #742: Provide a Jupyter magic function to get the status of running jobs
### What is the expected enhancement?
Get a way of inspecting the state of executed jobs by using a Jupyter magic function.

### PR DESCRIPTION
**Pull request by nonhermitian**
Fix #742

### Summary

Async job check for a qiskit job in Jupyter notebook env.

### CODE REVIEW

**Comment by delapuente** [comment]
To ease tracking, please remember to create an issue and making the PR to "Fix #issue" to keep both, problem and solution, related.

**Comment by nonhermitian** [comment]
@ajavadia, @ewinston  This has been updated to the new job format, please take a look.  How we test Jupyter extensions is still up in the air, but the current function seems to act appropriately.

**Comment by nonhermitian** [comment]
A line magic is a good idea.  I was thinking that one could not nest line magics, but it turns out your can.  Of course, you can have both.  I will just add the line magic.

**Comment by nonhermitian** [comment]
Ok, I decided that I did not like the feel of the line magic.  Instead I got around to making the cell magic work for multiple jobs in a single cell.  I think it is cleaner to have the magic and the `execute` call on separate lines.  Also realized that I was missing a check for when a job is cancelled.

File: qiskit/wrapper/jupyter/__init__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
```
Comment (diego-plan9) [inline qiskit/wrapper/jupyter/__init__.py]: Can you make sure the new files' headers follows the structure of the rest of the files of the project? It's important that we are consistent, and order and newlines do matter:

```
# -*- coding: utf-8 -*-

# Copyright 201x, IBM.
#
# This source code is licensed under the Apache License, Version 2.0 found in
# the LICENSE.txt file in the root directory of this source tree.

# pylint: disable=missing-docstring,redefined-builtin

"""Module blah for foo and bar."""

import ...
```

File: qiskit/wrapper/jupyter/__init__.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# pylint: disable=import-error
```
Comment (diego-plan9) [inline qiskit/wrapper/jupyter/__init__.py]: This one seems legit :tada:. However, can you move it to the lines where the jupyter/ipython imports are invoked, to ensure the scope of the disable is limited?

```
from IPython import get_ipython  # pylint: disable=import-error
from .jupyter_magics import StatusMagic
```

File: qiskit/wrapper/jupyter/__init__.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""Initilize the Jupyter routines.
+"""
+
+from IPython import get_ipython
+from .jupyter_magics import StatusMagic
+
+_ip = get_ipython()  # pylint: disable=C0103
```
Comment (diego-plan9) [inline qiskit/wrapper/jupyter/__init__.py]: Please remove this disable.

File: qiskit/wrapper/jupyter/jupyter_magics.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        type=float,
+        default=2,
+        help='Interval for status check.'
+    )
+    def qiskit_job_status(self, line='', cell=None):  # pylint: disable=W0613
+        """A Jupyter magic function to check the status of a Qiskit job instance.
+        """
+        args = magic_arguments.parse_argstring(self.qiskit_job_status, line)
+        # Split cell lines to get LHS variables
+        _cell_lines = cell.split('\n')
```
Comment (diego-plan9) [inline qiskit/wrapper/jupyter/jupyter_magics.py]: Can you change the variable names in this function so they do not start with an underscore? Variables declared inside a function have the scope of that function already, and treating them as private by prepending the `_` is not needed.

File: qiskit/wrapper/jupyter/jupyter_magics.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=import-error
```
Comment (diego-plan9) [inline qiskit/wrapper/jupyter/jupyter_magics.py]: Please see the note for this disable on the other file.

File: qiskit/wrapper/jupyter/jupyter_magics.py [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    @cell_magic
+    @magic_arguments.magic_arguments()
+    @magic_arguments.argument(
+        '-i',
+        '--interval',
+        type=float,
+        default=2,
+        help='Interval for status check.'
+    )
+    def qiskit_job_status(self, line='', cell=None):  # pylint: disable=W0613
```
Comment (diego-plan9) [inline qiskit/wrapper/jupyter/jupyter_magics.py]: Can you move this disable to the end of the docstring, for consistency with the rest of the project, and also use the human-readable-name:

```
"""A Jupyter magic function to check the status of a Qiskit job instance."""
# pylint: disable=some-name-that-can-be-understood
```

File: qiskit/wrapper/jupyter/__init__.py (hunk 1 above)
Comment (nonhermitian) [reply qiskit/wrapper/jupyter/__init__.py]: done

**Comment by nonhermitian** [comment]
The requested changes have been implemented.

**Comment by nonhermitian** [comment]
This needs one more mod before merging.

**Comment by nonhermitian** [comment]
This should now be done.

**Comment by nonhermitian** [comment]
@diego-plan9, please take a look at this again, and remove the merge block if satisfied.

**Comment by diego-plan9** [comment]
It's looking better, thanks @nonhermitian ! As mentioned in https://github.com/Qiskit/qiskit-terra/pull/734#pullrequestreview-150498092, can you attach (in the comments, not as a file in the codebase) a minimal jupyter notebook demonstrating the functionality? Since the PR is lacking tests (understandably), it would act as a sort of not-too-highly-automated test for this feature.

**Comment by diego-plan9** [comment]
Thanks @nonhermitian !

**Pull request merged**